frontend/web/components/TestWebhook.tsx

```diff
@@ -8,18 +8,72 @@ import Button from './base/forms/Button'
 type TestWebhookType = {
   webhook: string
   json: string
+  secret: string
 }
 
-const TestWebhook: FC<TestWebhookType> = ({ json, webhook }) => {
+// from https://stackoverflow.com/questions/24834812/space-in-between-json-stringify-output
+const stringifyWithSpaces = (str: string) => {
+  const obj = JSON.parse(str)
+  let result = JSON.stringify(obj, null, 1) // stringify, with line-breaks and indents
+  result = result.replace(/^ +/gm, ' ') // remove all but the first space for each line
+  result = result.replace(/\n/g, '') // remove line-breaks
+  result = result.replace(/{ /g, '{').replace(/ }/g, '}') // remove spaces between object-braces and first/last props
+  result = result.replace(/\[ /g, '[').replace(/ \]/g, ']') // remove spaces between array-brackets and first/last items
+  return result
+}
+
+const getSignature = (
+  body: string,
+  secret: string,
+  setSign: React.Dispatch<React.SetStateAction<string>>,
+) => {
+  const enc = new TextEncoder()
+  let str = ''
+
+  window.crypto.subtle
+    .importKey(
+      'raw',
+      enc.encode(secret),
+      {
+        hash: { name: 'SHA-256' },
+        name: 'HMAC',
+      },
+      false,
+      ['sign'],
+    )
+    .then((key) => {
+      window.crypto.subtle
+        .sign(
+          'HMAC',
+          key,
+          enc.encode(stringifyWithSpaces(body)), // We do this bc the python output is single line with one space before each value
+        )
+        .then((signature) => {
+          const b = new Uint8Array(signature)
+          str = Array.prototype.map
+            .call(b, (x) => x.toString(16).padStart(2, '0'))
+            .join('')
+          setSign(str)
+        })
+    })
+}
+
+const TestWebhook: FC<TestWebhookType> = ({ json, secret, webhook }) => {
   const [error, setError] = useState<string | null>(null)
   const [loading, setLoading] = useState(false)
   const [success, setSuccess] = useState(false)
+  const [sign, setSign] = useState('')
+  getSignature(json, secret, setSign)
+  const headers = {
+    'X-Flagsmith-Signature': sign,
+  }
+
   const submit = () => {
     setError(null)
     setLoading(true)
     setSuccess(false)
     data
-      .post(webhook, JSON.parse(json), null)
+      .post(webhook, JSON.parse(json), headers)
```

Comment:
RE above comment, this could be something more like 
```
getSignature().then((sign)=>{
 const headers = {
    'X-Flagsmith-Signature': sign,
  }
  data.post(...)
}
})
```

Reply:
Thanks for the review! Fixed at https://github.com/Flagsmith/flagsmith/compare/446e37228ee813f6f6503e5d2b950850f3b593a6..39cc696750aa1916fe6376fc6babbae6676cac0b along with a bit of refactoring